### PR TITLE
UnrealBuildTool parameters fix for UE4 4.18 on Linux

### PR DESCRIPTION
**Pull request by ashleysommer**
While trying to use this plugin on VSCode 1.17.2 on linux, with Unreal Editor 4.18-preview 4, I ran into a couple of errors.

Firstly, when generating project files the buildtool helper added arguments with "-project" and the actual project string as separate commands. The output of this was an argument with a space in it, for example:
```
mono /path/to/UnrealBuildTool.exe -makefile -cmakefile -codelitefile -project= /path/to/my/project/Project.uproject -game -rocket -waitmutex -engine
```
The `-project= /path/to/my/project/Project.uproject` is getting interpreted as two different arguments, then UnrealBuildTool returns a error stating it cannot find the target. Removing this space buy putting the `-project=` and `info.projectFilePath` concatenated onto one argument fixes this, and it works fine. I also made a slight tweak to this, to surround the projectFilePath with quotation marks, in case the project path has spaces in it.

The second problem, was when `getGenerateProjectFileArgs` is configured to _only_ generate codeLite project files, it was generating a command like this:
```
mono /path/to/UnrealBuildTool.exe -codelitefile -project=/path/to/my/project/Project.uproject -game -rocket -waitmutex -engine
```
When running this, I was getting "ERROR: Couldn't find platform name."
After some experimentation, I determined that on linux you cannot pass _just_ `-codelitefile`, you must also have `-makefile` as a minimum. So I add that in if it is not present already.

With these two changes, this plugin is working great on VSCode 1.17.2 with Unreal Engine 4.18-preview4.

### CODE REVIEW

**Comment by christopherreed** [comment]
Thank you for taking the time to put together this pull request and sorry for my delayed response!

I've been hesitant to update this extension for 4.18 since UE4 will ship with official support for VSCode in that version. I'm not sure how much, if any, additional value this extension will be to 4.18 users. UE4's official VSCode support should cover all its current functionality (code completion, build tasks, and debugging).

I feel like additional functionality I had been intending to work on for this project is probably better suited to being a seperate extension at this point.

I did, very briefly, investigate getting this extension up and running with 4.18 and ran into the issues your pull request addresses.

Regarding the first issue with `-project=` and `/path/to/your/project` being seperate args, that was a bit of laziness on my part I had intended to clean up ;) I should re-check but I'm pretty sure it worked OK with 4.16 and 4.17 on both Linux and Windows.

I suspect the cause of both the errors you fix is related to changes with the args to Unreal Build Tool for generating project files in 4.18. I don't fully remember the details but I think they made changes along these lines:
1) You now specify `-projectfiles` to generate the default project files for your platform (previously this arg was only used on Windows to generate the Visual Studio project files)
2) You now specify `/path/to/your/project.uproject` without the `-project=`

For example, I think the intended way to generate the project's Makefile and the official vscode project files in 4.18 is now something like:
`/path/to/UnrealBuildTool.exe -makefiles -vscode /path/to/project/project.uproject -game -engine`

Checking the engine version and implementing these fixes wouldn't be very difficult, but I'm currently leaning towards just supporting 4.16 and 4.17.

I'll leave this PR open until I have a bit more time to test and make a decision regarding 4.18 support.

Thanks!

**Comment by ashleysommer** [comment]
> For example, I think the intended way to generate the project's Makefile and the official vscode project files in 4.18 is now something like:
`/path/to/UnrealBuildTool.exe -makefiles -vscode /path/to/project/project.uproject -game -engine`

You're absolutely right. A couple of hours after I made this PR, I came across these changes when looking at the command line output of UE4Editor. I haven't made any further modifications to the plugin code to see if I can get it working with the new syntax, because it is working great for me now with just these changes.

>I've been hesitant to update this extension for 4.18 since UE4 will ship with official support for VSCode in that version. I'm not sure how much, if any, additional value this extension will be to 4.18 users. UE4's official VSCode support should cover all its current functionality (code completion, build tasks, and debugging).

I had the same thought, and was googling for some answers when I came across this support post:
https://answers.unrealengine.com/questions/708864/using-vscode-with-418-preview-1.html

Basically, in the 4.18 preview releases, VSCode support is _worse_ than it was in 4.17. It will probably get better on 4.18 final release, but for now we need this plugin :)

**Comment by christopherreed** [comment]
I pushed a hotfix release to address changes to Unreal Build Tool in 4.18.

You will need to manually add `"ue4-cpptools.engineVersion":"4.18"` to your settings. The engine version won't be detected automatically at this time.

> After some experimentation, I determined that on linux you cannot pass just `-codelitefile,` you must also have `-makefile` as a minimum.

This behavior appears to be somehow related to the Code Lite arg being renamed to `-codelitefiles`.

> Firstly, when generating project files the buildtool helper added arguments with "-project" and the actual project string as separate commands.

I might still need to fix this. Linux seems to work OK with 4.17.2 and 4.18.1, but I haven't had a chance to re-check on Windows yet.

Let me know if you still have issues.
